posts/2023-01-20-types-announcement.md

```diff
@@ -0,0 +1,118 @@
+
+There are a few reasons why we're sharing this post now (as opposed to months before or...never). First, the team finished a three day in-person/hybrid meetup at the beginning of December and we'd like to share the purpose and outcomes of that meeting. Second, posting this announcement now is just around 7 months of activity and we'd love to share what we've accomplished within this time. Lastly, as we enter into the new year of 2023, it's a great time to share a bit of where we expect to head in this year and beyond.
+
+## Background - How did we get here?
+
+Rust has grown significantly in the last several years, in many metrics: users, contributors, features, tooling, documentation, and more. As it has grown, the list of *things* people want to do with it has grown just as quickly. On top of powerful and ergonomic features, the demand for powerful tools such as IDEs or learning tools for the language has become more and more apparent. New compilers (frontend and backend) are being written. And, to top it off, we want Rust to continue to maintain one of its core design principles: safety.
+
+All of these points highlights some key needs: to be able to *know* how the Rust language should work, to be able to *extend* the language and compiler with new features in a relatively painless way, to be able to *hook into* the compiler and be able to query important information about programs, and finally to be able to *maintain* the language and compiler in an amenable and robust way. Over the years, considerable effort has been put into these needs, but we haven't *quite* achieved these key requirements.
+
+To extend a little, and put some numbers to paper, there are currently around 220 open tracking issues for [language](https://github.com/rust-lang/rust/issues?q=is%3Aopen+is%3Aissue+label%3AC-tracking-issue+label%3AT-lang), [compiler](https://github.com/rust-lang/rust/issues?page=1&q=is%3Aopen+is%3Aissue+label%3AC-tracking-issue+label%3AT-compiler), or [types](https://github.com/rust-lang/rust/issues?q=is%3Aopen+is%3Aissue+label%3AC-tracking-issue+label%3AT-types) features that have been accepted but are not completely implemented, of which about half are at least 3 years old and many are several years older than that. This is on top of the "deferred" RFCs that were never accepted: not because the idea itself wasn't appealing, but because of the lack of bandwidth to see the features through to stabilization. It's clear that we still need better foundations for making changes to the language and compiler.
```

Comment:
This still feels a little strange to me. I feel like bandwidth is a problem, but not related to T-types; unless you're saying that fixing the unsoundness will make it easier to implement new features because the refactors make the code simpler? But I'm not sure the code itself is the main barrier, I feel like contributor time (especially reviewer time) is the main issue ...

Reply:
> unless you're saying that fixing the unsoundness will make it easier to implement new features because the refactors make the code simpler

The causality here is a bit off, but the sentiment is close: Many of the current types-related tracking issues are open not solely because of bandwidth, but because working on the features is *hard*, in large part because putting the relevant semantics in context of the larger language *properly* is hard. Both the planned refactorings *and* the formalization of the type system should make the addition of new features more feasible (both in design and implementation). Fixing unsoundness issues is a separate, but related goal.

Sure, actual time is always an issue, but I think a bigger problem is many of these problems have sat for so long because it's not easy for anyone to take a look at them and know what needs to be done to finish them.

Reply:
👍 makes sense - maybe put that last sentence in the blog post directly?

Reply:
I tried to work the sentence and context in. Thoughts?

Reply:
It looks good to me :) thank you!